lib/src/lifecycle_module.dart

```diff
@@ -80,12 +106,30 @@ abstract class LifecycleModule {
   /// Public method to async load a child module and register it
   /// for lifecycle management.
   Future loadChildModule(LifecycleModule newModule) {
+    if (_childModules.contains(newModule)) {
```

Comment:
This keeps us from listening to the child's event streams more than once and ensures that the subscriptions get fully canceled when the child module unloads.